### ISSUE TITLE
Magnum Health Check wont work properly 

### ISSUE THREAD

**Issue by nguyenhuukhoi** [vexxhost/magnum-cluster-api#263]
Hello.
I am testing the Auto Health function. My cluster including 3 masters and 3 workers then I remove 3 workers. It is ok, new workers have joined completely but Health Status Reason wont update new workers, It still show old ones.

Environment:
Openstack Yoga
Magnum Cluster API  0.12.0
Cluster Management: v1.4.8

Thank you.

**Comment by ngyenhuukhoi** [comment]
I see that "NAME" and "NODENAME" when I run kubectl get machines -A are different. 
Does it make user mismatch when checking Health Status Reason? It will not show worker name in Opesntack but just name from kubectl get machines -A.

**Comment by mnaser** [comment]
I believe that in the name of nodes that you see there, we're using the machine and not the OpenStackMachine at the end of the day.

This is just impacting a "presentation" layer, as I understand, things are working correctly right?

**Comment by ngyenhuukhoi** [comment]
Yes, it is working correctly. I think user will confuse if they check base on "Machine". As a user viewpoint, I may ask about it.
